### PR TITLE
Support PHP 7.4 and 8.0

### PR DESCRIPTION
**Pull request by GrahamCampbell**
| Q               | A
| --------------- | ---
| Bug fix?        | yes
| New feature?    | yes
| BC breaks?      | no
| Deprecations?   | no
| License         | MIT

EDIT: Passing https://github.com/GrahamCampbell/mock-client/actions/runs/167569254.

### CODE REVIEW

**Comment by GrahamCampbell** [comment]
// cc @dbu ;0

File: spec/ClientSpec.php [hunk 1]
```diff
@@ -7,7 +7,7 @@
 use Http\Message\RequestMatcher;
 use Http\Message\ResponseFactory;
 use Http\Mock\Client;
-use Psr\Http\Client\ClientExceptionInterface;
+use Http\Client\Exception as ClientExceptionInterface;
```
Comment (dbu) [inline spec/ClientSpec.php:10]: why is this change needed? is the mock client picky about what kind of exception we can give it?

Comment (GrahamCampbell) [reply spec/ClientSpec.php:10]: Yes, it raises a deprecation warning, and phpspec's error handler doesn't hide it properly on PHP 8.0.

Comment (dbu) [reply spec/ClientSpec.php:10]: ah, ok. guess its better to test the recommended way of doing this rather than the deprecated anyways :+1:
